docs/src/Provider.doc.js

```diff
@@ -77,7 +90,297 @@ function Example(props) {
     </Provider>
   );
 }`}
-  />,
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="On Navigation"
+      description="This example illustrates the implementation of `onNavigation` context to control the link functionality of Gestalt components externally. This example has 4 relevant parts: a Provider, an `onNavigation` high-order function, consumer components (Link, Button, IconButton, TapArea), and `onNavigationOptions` props. The top Provider passes the `onNavigation` function to consumer components. Then, `onNavigation` returns a function that gets called during the `onClick` event handler. The `onNavigation` function can contain complex logic including React hooks to perform side effects. In this case, `onNavigation` is used to disable the link, show an alert message, and open a different URL in a new window. Finally, the `onNavigationOptions` prop provides a flexible API. In this case, `navigationMode` allows to disable/enable the `onNavigation` logic. Other uses could be accessing `event.stopPropagation`."
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+function OnNavigation() {
+  const [clientOnNavigationMode, setClientOnNavigationMode] = React.useState(true);
+
+  const onNavigation = ({ href, onNavigationOptions }) => {
+
+    const onNavigationClick = ({ event }) => {
+      event.nativeEvent.preventDefault();
+      // eslint-disable-next-line no-alert
+      alert("Disabled link. Opening help.pinterest.com instead.");
+      window.open("https://help.pinterest.com", "_blank");
+    }
+
+    return onNavigationOptions && onNavigationOptions.navigationMode === "client"
+      ? onNavigationClick
+      : null;
+  }
+
+  const linkProps = {
+    href: "https://pinterest.com",
+    onNavigationOptions: {
+      navigationMode: clientOnNavigationMode ? "client" : "server"
+    },
+    target: "blank",
+  }
+
+  return (
+    <Provider onNavigation={onNavigation}>
+      <Flex direction="column" gap={2}>
+        <Flex direction="row" gap={2}>
+          <Switch
+            onChange={() => setClientOnNavigationMode(!clientOnNavigationMode)}
+            id="disable-buttons"
+            switched={clientOnNavigationMode}
+          />
```

Comment:
Could we use a radio button group in these examples instead? I'm not sure the Switch is very readable here 
![Uploading Screen Shot 2021-02-05 at 2.00.30 PM.png…]()

Reply:
DONE. REDID THE NAMING. IT WAS INCORRECT